adapters/repos/db/file_structure_migration.go

```diff
@@ -118,3 +132,78 @@ func makeNewRelPath(oldPath, prefix string) (newPath string) {
 	}
 	return
 }
+
+func (db *DB) migratePQVectors(plan migrationPlan) error {
+	for _, parts := range plan {
+		oldClassPath := path.Join(
+			db.config.RootPath,
+			toTitleCase(parts[0].index),
```

Comment:
Not sure if I understand it correctly - is this attempt of getting back original className based on lowercased value? If so I'm afraid that may be failing point.

Given the `classname`, original name might have been `ClassName`, `Classname`, `CLASSNAME`, `ClAsSnAmE` etc.
Maybe we could try to find old paths other way round - check which of existing directories matches (after lowercasing) known indexes, and then move files in old path to new path.